### ISSUE TITLE
Enabling `Lint/MixedCaseRange` causes `Style/RedundantRegexpEscape` to skip infractions

### ISSUE THREAD

**Issue by Drowze** [rubocop/rubocop#13826]
## Expected behavior

I expect that enabling `Style/RedundantRegexpEscape` will work the same regardless of what other cops are enabled.

## Actual behavior

Enabling `Lint/MixedCaseRange` seems to affect how `Style/RedundantRegexpEscape` detect infractions.

## Steps to reproduce the problem

* test.rb:
```ruby
PASS_REGEXP = %r{^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#$%^&*()_+={}\[\]:;"'<>,.?/\\|`~\-]).{8,}$}
```
* cop enabled alone detects infraction:
```
$ rubocop --only Style/RedundantRegexpEscape test.rb
Offenses:

test.rb:3:96: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
PASS_REGEXP = %r{^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#$%^&*()_+={}\[\]:;"'<>,.?/\\|`~\-]).{8,}$}
                                                                                               ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

* cop enabled with other cop doesn't detect infraction:
```
$ rubocop --only Lint/MixedCaseRange,Style/RedundantRegexpEscape test.rb
1 file inspected, no offenses detected
```

![Image](https://github.com/user-attachments/assets/f3a6695b-956f-48c9-94ed-ef7b80cdb63c)

## RuboCop version

```
$ rubocop -V
1.71.2 (using Parser 3.3.7.1, rubocop-ast 1.38.0, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin23]
```
